lib/chef/provider/package/yum.rb

```diff
@@ -1090,10 +1097,18 @@ def load_current_resource
 
           Chef::Log.debug("#{@new_resource} checking yum info for #{new_resource}")
 
-          installed_version = @yum.installed_version(@new_resource.package_name, arch)
-          @current_resource.version(installed_version)
-
-          @candidate_version = @yum.candidate_version(@new_resource.package_name, arch)
+          installed_version = []
+          @candidate_version = []
+          package_name_array.each do |pkg|
+            installed_version << @yum.installed_version(pkg, arch)
+            @candidate_version << @yum.candidate_version(pkg, arch)
+          end
+          #if installed_version.size == 1
```

Comment:
the commented out code? Yes, absolutely, this was leftover from the conversion. It's unclear that keeping package_name as an array always is the right call here (as opposed to always using the `*_array` helpers)... especially because we cannot do it for package_name on the new resource...